lms/djangoapps/discussion_api/api.py

```diff
@@ -39,6 +41,33 @@ def _get_course_or_404(course_key, user):
     return course
 
 
+def _get_thread_and_context(request, thread_id, parent_id=None, retrieve_kwargs=None):
+    """
+    Retrieve the given thread and build a serializer context for it, returning
+    both. This function also enforces access control for the thread (checking
+    both the user's access to the course and to the thread's cohort if
+    applicable). Raises Http404 if the thread does not exist or the user cannot
+    access it.
+    """
+    retrieve_kwargs = retrieve_kwargs or {}
+    try:
+        if "mark_as_read" not in retrieve_kwargs:
+            retrieve_kwargs["mark_as_read"] = False
+        cc_thread = Thread(id=thread_id).retrieve(**retrieve_kwargs)
+        course_key = CourseLocator.from_string(cc_thread["course_id"])
```

Comment:
i think this should be CourseKey instead of CourseLocator (source: [this wiki page](https://github.com/edx/edx-platform/wiki/Opaque-Keys-%28Locators%29#utilizing-keys))